hawkular-alerts-rest/src/main/java/org/hawkular/alerts/rest/EventsHandler.java

```diff
@@ -97,7 +96,11 @@ public Response createEvent(
                 if (null != alertsService.getEvent(tenantId, event.getId(), true)) {
                     return ResponseUtil.badRequest("Event with ID [" + event.getId() + "] exists.");
                 }
-                alertsService.addEvents(Collections.singletonList(event));
+                /*
+                    New events are sent directly to the engine for inference process.
+                    Input events and new ones generated by the alerts engine are persisted at the end of the process.
+                 */
+                alertsService.sendEvent(event);
```

Comment:
Or renaming/accomodate wording but in essence what does sendEvent() is what it should be public.